test/pytest/test_handler_traceback_logging.py

```diff
@@ -0,0 +1,193 @@
+        no_config_snapshots=True,
+        gen_mar=False,
+        snapshot_file=config_properties_file.as_posix(),
+    )
+
+    yield pipe
+
+    test_utils.torchserve_cleanup()
+
+
+@pytest.fixture(scope="module", name="mar_file_path")
+def create_mar_file(work_dir, model_archiver, model_name):
+    mar_file_path = work_dir.joinpath(model_name + ".mar")
+
+    model_py_file = work_dir / "model.py"
+    model_py_file.write_text(MODEL_PY)
+
+    model_config_yaml_file = work_dir / "model_config.yaml"
+    model_config_yaml_file.write_text(MODEL_CONFIG_YAML)
+
+    handler_py_file = work_dir / "handler.py"
+    handler_py_file.write_text(HANDLER_PY)
+
+    config = ModelArchiverConfig(
+        model_name=model_name,
+        version="1.0",
+        serialized_file=None,
+        model_file=model_py_file.as_posix(),
+        handler=handler_py_file.as_posix(),
+        extra_files=None,
+        export_path=work_dir,
+        requirements_file=None,
+        runtime="python",
+        force=False,
+        archive_format="default",
+        config_file=model_config_yaml_file.as_posix(),
+    )
+
+    with patch("archiver.ArgParser.export_model_args_parser", return_value=config):
+        model_archiver.generate_model_archive()
+
+        assert mar_file_path.exists()
+
+        yield mar_file_path.as_posix()
+
+    # Clean up files
+    mar_file_path.unlink(missing_ok=True)
+
+
+@pytest.fixture(scope="module", name="model_name")
+def register_model(mar_file_path, model_store, torchserve):
+    """
+    Register the model in torchserve
+    """
+    shutil.copy(mar_file_path, model_store)
+
+    file_name = Path(mar_file_path).name
+
+    model_name = Path(file_name).stem
+
+    params = (
+        ("model_name", model_name),
+        ("url", file_name),
+        ("initial_workers", "1"),
+        ("synchronous", "false"),
+        ("batch_size", "1"),
+    )
+
+    test_utils.reg_resp = test_utils.register_model_with_params(params)
+
+    yield model_name, torchserve
+
+    test_utils.unregister_model(model_name)
+
+
+@pytest.mark.timeout(60)
+def test_handler_traceback_logging(model_name):
+    """
+    Full circle test with torchserve
+    """
+
+    model_name, pipe = model_name
+
+    traceback = [
+        "Traceback (most recent call last):",
+        "line 12, in initialize",
+        "self.call1()",
+        "line 18, in call1",
+        "self.call2()",
+        "line 21, in call2",
+        "self.call3()",
+        "line 24, in call3",
+        "self.call4()",
+        "line 27, in call4",
+        "self.call5()",
+        "line 30, in call5",
+        "assert False",
+        "AssertionError",
+    ]
+
+    # Test traceback logging for first attempt and three retries to start worker
+    for _ in range(4):
+        logs = []
+        while True:
+            logs.append(pipe.get())
+            if "AssertionError" in logs[-1]:
+                break
+
+        for line in traceback:
+            assert any(line in log for log in logs)
```

Comment:
I think this should be:
```
assert all(line in logs for line in traceback)
```
any will stop after the first line is found

Reply:
`line in logs` would not work because the logs include other details such as timestamp and log level along with the traceback line which cannot be hardcoded in the test, so I believe the logic would have to be `line in log for log in logs`.

Agree that `any will stop after the first line is found`, which is why I loop over each of the traceback lines I expect to find the captured logs:

```
        for line in traceback:
            assert any(line in log for log in logs)
```